apps/hexagon_matmul/pipeline.cpp

```diff
@@ -0,0 +1,112 @@
+#include "Halide.h"
+
+using namespace Halide;
+using namespace Halide::ConciseCasts;
+
+int main(int argc, char **argv) {
```

Comment:
i was kind of eager to try this patch  .

   i see an abort on conv3x3a16 and conv3x3a32 and helloHexagon:
       
Internal error at /local/mnt2/ronl/Halide-upstream/Halide/src/IR.cpp:21
Condition failed: a.type() == b.type()
Add of mismatched types